src/exabgp/reactor/api/command/rib.py

```diff
@@ -23,70 +25,88 @@
+                                        "local-ip": str(neighbor_data['local-ip']),
+                                        "local-as": neighbor_data['local-as'],
+                                        "peer-as": neighbor_data['peer-as'],
+                                        "router-id": str(neighbor_data['router-id']),
+                                        "family-allowed": neighbor_data['family-allowed'],
+                                    })
+
+                            route_entry = {
+                                "prefix": str(change.nlri.cidr.prefix()),
+                                "family": str(change.nlri.family()).strip("()").replace(",", "")
+                            }
+                            output[neighbor_ip]["routes"].append(route_entry)
                         else:
-                            reactor.processes.write(
-                                service,
-                                'neighbor %s %s %s'
-                                % (
-                                    reactor.neighbor_ip(key),
-                                    '%s %s' % change.nlri.family().afi_safi(),
-                                    str(change.nlri),
-                                ),
-                            )
+                            if extensive:
+                                reactor.processes.write(
+                                    service,
+                                    '%s %s %s'
+                                    % (reactor.neighbor_name(key), '%s %s' % change.nlri.family(), change.extensive()),
+                                )
+                            else:
+                                reactor.processes.write(
+                                    service,
+                                    'neighbor %s %s %s'
+                                    % (reactor.neighbor_ip(key), '%s %s' % change.nlri.family(), str(change.nlri)),
+                                )
                 yield True
+
+        if json_output:
+            reactor.processes.write(service, json.dumps(output, indent=4))
+
         reactor.processes.answer_done(service)
 
     return callback
 
 
-@Command.register(
-    'text',
-    'show adj-rib out',
-    False,
-    [
-        'extensive',
-    ],
-)
-@Command.register(
-    'text',
-    'show adj-rib in',
-    False,
-    [
-        'extensive',
-    ],
-)
+@Command.register('text', 'show adj-rib out', False, ['extensive',])
+@Command.register('text', 'show adj-rib in', False, ['extensive',])
+@Command.register('text', 'show adj-rib out json', False, ['extensive',])
+@Command.register('text', 'show adj-rib in json', False, ['extensive',])
 def show_adj_rib(self, reactor, service, line):
     words = line.split()
     extensive = line.endswith(' extensive')
+    json_output = True if 'json' in words else False
     try:
         rib = words[2]
-        if not rib in ('in', 'out'):
+        if not rib in ('in', 'out', 'json'):
```

Comment:
This logic seems wrong ... I would need to look more in depth to say why but I can not ATM.

Reply:
For clarity it is also about the `if rib == 'json'` part too.